manywheel/build_rocm.sh

```diff
@@ -223,84 +229,108 @@ HIPBLASLT_LIB_FILES=($ARCH_SPECIFIC_FILES $OTHER_FILES)
-DEPS_SONAME=(
-    ${ROCM_SO_FILES[*]}
-)
-if [[ "$BUILD_LIGHTWEIGHT" != "1" ]]; then
-    DEPS_SONAME+=(${OS_SO_FILES[*]})
-fi
+    # Set environment so build_common.sh (or build_libtorch.sh) sees it
+    DEPS_LIST=( "${ROCM_SO_PATHS_LIGHTWEIGHT[@]}" )
+    DEPS_SONAME=( "${LIGHTWEIGHT_ROCM_SO_FILES[@]}" )
+    DEPS_AUX_SRCLIST=()
+    DEPS_AUX_DSTLIST=()
 
-DEPS_AUX_SRCLIST=()
-if [[ "$BUILD_LIGHTWEIGHT" != "1" ]]; then
-    DEPS_AUX_SRCLIST+=("${ROCBLAS_LIB_FILES[@]/#/$ROCBLAS_LIB_SRC/}")
-    DEPS_AUX_SRCLIST+=("${HIPBLASLT_LIB_FILES[@]/#/$HIPBLASLT_LIB_SRC/}")
-    DEPS_AUX_SRCLIST+=("/opt/amdgpu/share/libdrm/amdgpu.ids")
-fi
+    # Because original script doesn't add OS libraries or arch-specific .co
+    # files for "lightweight", we skip them here
 
-DEPS_AUX_DSTLIST=()
-if [[ "$BUILD_LIGHTWEIGHT" != "1" ]]; then
-    DEPS_AUX_DSTLIST+=("${ROCBLAS_LIB_FILES[@]/#/$ROCBLAS_LIB_DST/}")
-    DEPS_AUX_DSTLIST+=("${HIPBLASLT_LIB_FILES[@]/#/$HIPBLASLT_LIB_DST/}")
-    DEPS_AUX_DSTLIST+=("share/libdrm/amdgpu.ids")
-fi
-if [[ "$BUILD_LIGHTWEIGHT" != "1" ]]; then
-    if [[ $ROCM_INT -ge 50500 ]]; then
-        # MIOpen library files
-        MIOPEN_SHARE_SRC=$ROCM_HOME/share/miopen/db
-        MIOPEN_SHARE_DST=share/miopen/db
-        MIOPEN_SHARE_FILES=($(ls $MIOPEN_SHARE_SRC))
-
-        DEPS_AUX_SRCLIST+=(${MIOPEN_SHARE_FILES[@]/#/$MIOPEN_SHARE_SRC/})
-        DEPS_AUX_DSTLIST+=(${MIOPEN_SHARE_FILES[@]/#/$MIOPEN_SHARE_DST/})
+    # Finally, source the main build script
+    SCRIPTPATH="$( cd "$(dirname "$0")" ; pwd -P )"
+    if [[ -z "$BUILD_PYTHONLESS" ]]; then
+        BUILD_SCRIPT=build_common.sh
+    else
+        BUILD_SCRIPT=build_libtorch.sh
     fi
+    source "$SCRIPTPATH/${BUILD_SCRIPT}"
 
-    if [[ $ROCM_INT -ge 50600 ]]; then
-        # RCCL library files
-        if [[ $ROCM_INT -ge 50700 ]]; then
-            RCCL_SHARE_SRC=$ROCM_HOME/share/rccl/msccl-algorithms
-            RCCL_SHARE_DST=share/rccl/msccl-algorithms
-        else
-            RCCL_SHARE_SRC=$ROCM_HOME/lib/msccl-algorithms
-            RCCL_SHARE_DST=lib/msccl-algorithms
+    echo "=== Done building LIGHTWEIGHT variant ==="
+}
+
+# 2) HEAVYWEIGHT BUILD
+do_heavyweight_build() {
+    echo "=== Building HEAVYWEIGHT variant ==="
+
+    # Gather file paths for the full set
+    ROCM_SO_PATHS_HEAVYWEIGHT=()
+    for lib in "${HEAVYWEIGHT_ROCM_SO_FILES[@]}"; do
+        file_path="$(find "$ROCM_HOME/lib" -name "$lib" 2>/dev/null | head -n1)"
+        if [[ -z $file_path && -d "$ROCM_HOME/lib64" ]]; then
+            file_path="$(find "$ROCM_HOME/lib64" -name "$lib" 2>/dev/null | head -n1)"
         fi
-        RCCL_SHARE_FILES=($(ls $RCCL_SHARE_SRC))
+        if [[ -z $file_path ]]; then
+            file_path="$(find "$ROCM_HOME" -name "$lib" 2>/dev/null | head -n1)"
+        fi
+        if [[ -z $file_path ]]; then
+            echo "Error: Heavyweight library $lib not found." >&2
+            exit 1
+        fi
+        ROCM_SO_PATHS_HEAVYWEIGHT+=("$file_path")
```

Comment:
This logic seems duplicated across the do_lightweight_build/do_heavyweight_build. Can we not use a common function for both?